File: src/main/java/com/matera/restserver/exception/EntityExistsException.java
package com.matera.restserver.exception;

import com.matera.restserver.util.Messages;

public class EntityExistsException extends Exception {

	private static final String DEFAULT_MESSAGE = "exceptions.default.entityexistsexception";

	private static final long serialVersionUID = -4802286605958048986L;

	public EntityExistsException() {
		super(Messages.getMessage(DEFAULT_MESSAGE));
	}

}

File: src/main/kotlin/com/matera/restserver/exception/EntityExistsException.kt
package com.matera.restserver.exception

import com.matera.restserver.util.Messages.getMessage

class EntityExistsException : Exception(getMessage(DEFAULT_MESSAGE)) {

    companion object {
        private const val DEFAULT_MESSAGE = "exceptions.default.entityexistsexception"
        private const val serialVersionUID = -4802286605958048986L
    }
}
